contracts/WETH10.sol

```diff
@@ -9,85 +9,64 @@ interface ERC677Receiver {
 }
 
 interface FlashMinterLike {
```

Comment:
i actually meant in my issue that this interface can be removed entirely

Reply:
And then what, call the fallback function as the target for flash minting? Am I missing something?

I find it safer that the receiving contract specifies that it accepts being the target of a flash mint by implementing `executeOnFlashMint`.

Reply:
you just native `call` the address with the input data. the input data can encode the function selector of the target contract in addition to all the arguments.

it's not any less safe to do it this way. e.g. i may call directly into the uniswap weth/dai pair to sell the weth as the target of the flash mint, and *then* from the swap call back into my own contract, having already swapped weth for some other asset. 

this is gas cheaper than transferring weth to my own contract first (implementing flash minter like interface) and then selling on uniswap which involves multiple transfers of weth.

uniswap may not be a good example given the use case is probably covered by flash swaps, but the point is the interface is not needed for safety

Reply:
Will be fixed in a separate PR